### DISCUSSION TITLE
Integration with quarto

### DISCUSSION THREAD

**Discussion by MerlinDumeur** [jeertmans/manim-slides#498] [Q&A]
Hi,

Thanks for developing this great tool for placing manim animations in slides. I am using quarto now to generate my reveal.js slide set, and I am wondering whether it is possible to use manim-slides to add manim animations to my slideshow: It seems like it should be possible but I can't seem to find any information on how to do that. Could you please point me to where I should start to achieve this?

**Comment by jeertmans** [comment]
Hi @MerlinDumeur, thanks for posting your question!

I am not familiar with Quarto, is it [this tool](https://quarto.org/)?

If so, you might probably want to check out the [IPython magic](https://manim-slides.eertmans.be/latest/reference/magic_example.html), as Quarto seems to be able to generate slides from Jupyter Notebooks, [right](https://quarto.org/docs/get-started/authoring/jupyter.html)?

**Reply by MerlinDumeur** [reply]
Hi @jeertmans, thanks for the swift reply!

> I am not familiar with Quarto, is it [this tool](https://quarto.org/)?
Yes it is that tool, sorry for being unclear.

> If so, you might probably want to check out the [IPython magic](https://manim-slides.eertmans.be/latest/reference/magic_example.html), as Quarto seems to be able to generate slides from Jupyter Notebooks, [right](https://quarto.org/docs/get-started/authoring/jupyter.html)?

Thanks, that seems like a good starting point.
The fundamental aim of quarto is to author every document in markdown and then use [pandoc](https://pandoc.org/) to render them in any framework: pptx, reveal.js, beamer, etc...

In that context, authoring from jupyter notebooks seems like it is internally getting translated to markdown first. Since I don't usually author documents in jupyterlab, it would be nice to figure out how to do it with any quarto markdown file.

I'm not afraid to get my hands dirty, if that means writing an extension for quarto for instance. I'll look into using jupyter first, then I'll report what I find :)

Thanks again!

**Reply by MerlinDumeur** [reply]
Since quarto exports directly from markdown to reveal.js, and manim-slides exports to reveal.js as well, I guess it should be possible to merge both via an extension to quarto in order to control animations in the quarto-generated reveal.js slides. Does that it look like that would make sense?

**Reply by jeertmans** [reply]
Yes merging slides would make sense, but this might not be easy as there is currently no easy way (I think) to identify some slide from Manim Slides. A long time ago, I thought of adding an `id` field to each slide, so you can later reference to it with RevealJS, for example.

If Quarto supports executing Python code prior to rendering, this might be the easiest solution.

Please let me know if you manage to have anything working!

**Reply by rodrigo-j-goncalves** [reply]
I give my lectures using [quarto](https://Quarto.org/) (format: revealjs), and I have always wanted to include manim animations created with manim-slides.

It is possible in Quarto to include a 'secondary' revealjs presentation inside a 'primary' one, but the presenter needs to transfer focus from the primary to the secondary by clicking on the secondary. Then you advance the 'secondary' slides, and need to click back outside the secondary presentation to return focus to the primary presentation and continue with the primary slides.

There is also an `AutoSlide` option (i.e., the 'secondary' presentation starts automatically after loading). Still, it is not useful for me because I want to control the flow and timing of each slide as my lecture/talk progresses (sometimes I even need to go back to previous slides).

The focus thing is quite tricky, and it was not easy to get around it.

I ended up creating a Quarto extension for this (I _claudeveloped_ it). Now I can `splice` several presentations without worrying about the focus.
And the _flow of slides_ is still a single, linear progression that I advance with the usual keys.

You can have a look here: [quarto splice](https://github.com/rodrigo-j-goncalves/splice/)

@MerlinDumeur, have a look. I hope you find it useful.

@jeertmans, if you think this could be advertised on the main-slides website, go ahead! Or give me instructions 😃   
And by the way, thank you so much for manim-slides!!!! 👏🏽
